src/entities/entityid.ts

```diff
@@ -0,0 +1,43 @@
+// tslint:disable:member-access
+
+import { Utils } from "../utils";
+
+/**
+ * A unique identifier for an entity, consisting of entity class and entity key.
+ */
+export class EntityId {
+    /** @hidden */
+    static getEntityIdFromSchedulerId(schedulerId: string): EntityId {
+        const pos = schedulerId.indexOf("@", 1);
+        const entityName = schedulerId.substring(1, pos);
+        const entityKey = schedulerId.substring(pos + 1);
+        return new EntityId(entityName, entityKey);
+    }
+
+    /** @hidden */
+    static getSchedulerIdFromEntityId(entityId: EntityId): string {
+        return `@${entityId.name}@${entityId.key}`;
+    }
+
+    /**
+     * Create an entity id for an entity.
+     */
+    constructor(
+        // TODO: consider how to name these fields more accurately without interfering with JSON serialization
+        /** The name of the entity class. */
+        public readonly name: string,
+        /** The entity key. Uniquely identifies an entity among all instances of the same class. */
+        public readonly key: string,
+    ) {
+        Utils.throwIfEmpty(name, "name");
+        Utils.throwIfEmpty(key, "key");
+    }
+
+    public toString(): string {
+        return EntityId.getSchedulerIdFromEntityId(this);
+    }
+
+    // TODO: equals
+
+    // TODO: comparer
```

Comment:
I don't think these are technically required (unlike C#), just nice to have for tests and potentially user code.

@kashimiz your thoughts here?

Reply:
It depends on whether you need them for implementation of any methods (ex. `IDurableOrchestrationContext.LockAsync`). At the time I made these notes I wasn't 100% sure of what certain methods would look like on the JS side.

Otherwise, I'd count them as nice-to-have.

Reply:
I think I will remove these TODOs, and add the methods as necessary.